src/framework/lib/genesisframework/task.rb

```
@@ -112,32 +112,51 @@ def install provider, *what
             # installed and just pass it to install anyway.
-            gems = what.select do |item|
-              item.include?("-") \
-                || item.include?("/") \
-                || Gem::Dependency.new(item).matching_specs.count == 0
+            install_gems = gems.select do |gem, requires|
+              gem.include?("-") \
+                || gem.include?("/") \
+                || Gem::Dependency.new(gem).matching_specs.count == 0
             end
-            if gems.size > 0    # make sure we still have something to do
-              Kernel.system('gem', 'install', '--no-ri', '--no-rdoc', *gems)
+
+            if install_gems.size > 0    # make sure we still have something to do
+              options = config.fetch(:gem_args, '').split
+              args = (options << install_gems.keys).flatten
+              Kernel.system('gem', 'install', *args)
               if $?.exitstatus != 0
-                raise "gem install #{gems.join(' ')} exited with status: " \
+                raise "gem install #{args.join(' ')} exited with status: " \
                   + $?.exitstatus.to_s
               end
             else                # be noisy that we aren't doing anything
-              puts "already installed gems: #{what.join(' ')}"
+              puts "already installed gems: #{gems.keys.join(' ')}"
             end
 
             # now need to clear out the Gem cache so we can load it
             Gem.clear_paths
 
-            # Now we require all the gems you asked to be installed
-            what.all? { |gem| require gem }
+            # Attempt to require loaded gems, print a message if we can't.
+            gems.each { |gem, requires|
+              begin
+                requires.each {|r| require r }
+              rescue LoadError
+                puts "Could not load gem #{gem} automatically. Maybe the gem name differs from its load path? Please specify the name to require, or load it explicitly in the task."
```

The idea was to let people require the gem themselves in the task if we failed to load it automatically, but I guess now that we provide a way to specify requires with `install :gem` we probably should raise an error as you say.